packages/identity/src/identity.ts

```diff
@@ -1,80 +1,75 @@
-  value: string,
-  appSignaturePublicKey: Uint8Array,
-  appEncryptionPublicKey: Uint8Array,
-  tankerSignaturePublicKey: Uint8Array,
-  tankerEncryptionPublicKey: Uint8Array,
-|};
-
-export type ProvisionalUserKeys = {|
-  appSignatureKeyPair: tcrypto.SodiumKeyPair,
-  appEncryptionKeyPair: tcrypto.SodiumKeyPair,
-  tankerSignatureKeyPair: tcrypto.SodiumKeyPair,
-  tankerEncryptionKeyPair: tcrypto.SodiumKeyPair,
-|};
+export type PublicPermanentIdentity = {
+  trustchain_id: b64string;
+  target: PermanentIdentityTarget;
+  value: b64string;
+};
+
+export type SecretPermanentIdentity = PublicPermanentIdentity & {
+  ephemeral_public_signature_key: b64string;
+  ephemeral_private_signature_key: b64string;
+  delegation_signature: b64string;
+  user_secret: b64string;
+};
+
+export type PublicProvisionalIdentity = {
+  trustchain_id: b64string;
+  target: ProvisionalIdentityTarget;
+  value: string;
+  public_signature_key: b64string;
+  public_encryption_key: b64string;
+};
+
+export type SecretProvisionalIdentity = PublicProvisionalIdentity & {
+  private_encryption_key: b64string;
+  private_signature_key: b64string;
+};
+
+export type PublicProvisionalUser = {
+  trustchainId: Uint8Array;
+  target: string;
+  value: string;
+  appSignaturePublicKey: Uint8Array;
+  appEncryptionPublicKey: Uint8Array;
+  tankerSignaturePublicKey: Uint8Array;
+  tankerEncryptionPublicKey: Uint8Array;
+};
+
+export type ProvisionalUserKeys = {
+  appSignatureKeyPair: tcrypto.SodiumKeyPair;
+  appEncryptionKeyPair: tcrypto.SodiumKeyPair;
+  tankerSignatureKeyPair: tcrypto.SodiumKeyPair;
+  tankerEncryptionKeyPair: tcrypto.SodiumKeyPair;
+};
 
 export type SecretIdentity = SecretPermanentIdentity | SecretProvisionalIdentity;
 export type PublicIdentity = PublicPermanentIdentity | PublicProvisionalIdentity;
 
-function isPermanentIdentity(identity: SecretIdentity | PublicIdentity): bool %checks {
+function isPermanentIdentity(identity: SecretIdentity | PublicIdentity): boolean {
   return identity.target === 'user';
 }
 
-function isPublicPermanentIdentity(identity: SecretPermanentIdentity | PublicPermanentIdentity): bool %checks {
+function isPublicPermanentIdentity(identity: SecretPermanentIdentity | PublicPermanentIdentity): boolean {
   return !('user_secret' in identity);
 }
 
-function isProvisionalIdentity(identity: SecretIdentity | PublicIdentity): bool %checks {
+function isProvisionalIdentity(identity: SecretIdentity | PublicIdentity): boolean {
   return identity.target === 'email';
 }
 
 export function _serializeIdentity(identity: SecretIdentity | PublicIdentity): b64string { // eslint-disable-line no-underscore-dangle
   return utils.toB64Json(identity);
 }
 
-function _deserializeAndFreeze(identity: b64string): Object { // eslint-disable-line no-underscore-dangle
-  const result = utils.fromB64Json(identity);
-
+function _deserializeAndFreeze(identity: b64string): SecretIdentity | PublicIdentity { // eslint-disable-line no-underscore-dangle
```

Comment:
```suggestion
function _deserializeAndFreeze(identity: b64string): SecretIdentity | PublicIdentity {
```

(maybe)

Reply:
```suggestion
function _deserializeAndFreeze(identity: b64string): Object { // eslint-disable-line no-underscore-dangle
```

We don't validate anything in the function, so I think it's still an Object, callers of this function will have to validate/ignore-cast to their desired type, but `_deserializeAndFreeze` really just does take any b64 string, parse it, and adds the raw input in a hidden field.